### PR TITLE
[tune] Fix `syncer=None` not disabling trial-to-driver syncing

### LINKED ISSUE #20361: [release][nightly] `result_throughput_cluster ` failed
https://buildkite.com/ray-project/periodic-ci/builds/1625#113bf939-dd99-4838-a0ca-583f219e79e8

### PR DESCRIPTION
**Pull request by krfricke**
## Why are these changes needed?

`syncer=None` did not disable trial checkpoint to driver syncing, which is resolved and tested with this PR.

## Related issue number

Closes #20361

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by gjoliver — APPROVED** [review]
just 1 minor suggestion. thanks.

File: python/ray/tune/syncer.py [hunk 1]
```diff
@@ -556,8 +556,8 @@ def detect_cluster_syncer(
 
     sync_config = sync_config or SyncConfig()
 
-    if bool(sync_config.upload_dir):
-        # No sync to driver for cloud checkpointing
+    if bool(sync_config.upload_dir) or sync_config.syncer is None:
```
[the remark this answers is not in the document]
Comment (krfricke) [reply python/ray/tune/syncer.py:559]: We also want to catch `upload_dir=None` here - usually I might do `if sync_config.upload_dir` which should evaluate to the same expression.
But I agree it's not ideal, I may refactor this when cleaning up the syncing tech debt

**Review by Yard1 — APPROVED** [review]
LGTM

**Comment by krfricke** [comment]
Release test passed: https://buildkite.com/ray-project/periodic-ci/builds/1653

**Comment by xwjiang2010** [comment]
Thanks Kai!

**Pull request merged**